### PR TITLE
feat(seldonimage): L3-4753 more opts

### PR DESCRIPTION
**Pull request by sgrund14**
**Jira ticket**

[L3-4753](https://phillipsauctions.atlassian.net/browse/L3-4753)

**Summary**

More image optimization logic in SeldonImage and memoizing object tile

**Change List (describe the changes made to the files)**

- More image opt logic in SeldonImage
- Memoize object tile

**Acceptance Test (how to verify the PR)**

- SeldonImage and ObjectTile still work

**Regression Test**

- (Optional) Add verification steps to make sure this PR doesn't break old functionality

**Evidence of testing**

- Post logs, screenshots, etc

**Things to look for during review**

- [ ] PR title should correctly describe the most significant type of commit. I.e. `feat(scope): ...` if a `minor` release should be triggered.
- [ ] All commit messages follow convention and are appropriate for the changes
- [ ] All references to `phillips` class prefix are using the prefix variable
- [ ] All major areas have a `data-testid` attribute.
- [ ] Document all props with jsdoc comments
- [ ] All strings should be translatable.
- [ ] Unit tests should be written and should have a coverage of 90% or higher in all areas.

New Components

- [ ] Are there any [accessibility considerations](https://www.w3.org/WAI/ARIA/apg/patterns/) that need to be taken into account and tested?
- [ ] Default story called "Playground" should be created for all new components
- [ ] Create a jsdoc comment that has an Overview section and a link to the Figma design for the component
- [ ] Export the component and its typescript type from the `index.ts` file
- [ ] Import the component scss file into the `componentStyles.scss` file.

[L3-4753]: https://phillipsauctions.atlassian.net/browse/L3-4753?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/patterns/ObjectTile/ObjectTile.tsx [hunk 1]
```diff
@@ -78,110 +86,116 @@ export interface ObjectTileProps extends ComponentProps<'a'> {
  *
  * [Storybook Link](https://phillips-seldon.netlify.app//?path=/story/patterns-objecttile--playground)
  */
-const ObjectTile = forwardRef<HTMLAnchorElement, ObjectTileProps>(
```
Comment (sgrund14) [inline src/patterns/ObjectTile/ObjectTile.tsx:81]: only change here is memoizing & passing new props to image

File: src/components/SeldonImage/SeldonImage.tsx [hunk 2]
```diff
@@ -91,16 +125,44 @@ const SeldonImage = memo(
...
+        if (isServer) {
+          return 'loading';
+        }
+        const element = document.getElementById(src);
+
+        if (element instanceof HTMLImageElement && !element.classList.contains(`${baseClassName}-img--hidden`)) {
+          return 'loaded';
+        }
+
+        const img = document.createElement('img');
```
Comment (scottdickerson) [inline src/components/SeldonImage/SeldonImage.tsx:138]: I'm a bit surprised you had to do this all over again and couldn't just do this in the `isImageValid`?

File: src/patterns/SaleHeaderBanner/SaleHeaderBanner.tsx [hunk 3]
```diff
@@ -67,6 +83,10 @@ const SaleHeaderBanner = forwardRef<HTMLDivElement, SaleHeaderBannerProps>(
       auctionEndTime,
       auctionTitle,
       imageSrcUrl,
+      imageSrcSet,
+      imageSizes,
+      imageLoading,
```
Comment (scottdickerson) [inline src/patterns/SaleHeaderBanner/SaleHeaderBanner.tsx:88]: should we have defaults here for Loading and Priority?

File: src/components/SeldonImage/SeldonImage.tsx [hunk 4]
```diff
@@ -83,6 +115,8 @@ const SeldonImage = memo(
         alt,
         srcSet,
         sizes,
+        loading,
```
Comment (scottdickerson) [inline src/components/SeldonImage/SeldonImage.tsx:118]: should we have defaults for these?

File: src/components/SeldonImage/SeldonImage.tsx (hunk 2 above)
Comment (sgrund14) [reply src/components/SeldonImage/SeldonImage.tsx:138]: tbh this part was just adding extra redundancy, probably could get away with removing it

File: src/patterns/SaleHeaderBanner/SaleHeaderBanner.tsx (hunk 3 above)
Comment (sgrund14) [reply src/patterns/SaleHeaderBanner/SaleHeaderBanner.tsx:88]: if we want, figured consumer can pass in if they want (the image itself has defaults to `auto` on each one)

File: src/components/SeldonImage/SeldonImage.tsx (hunk 4 above)
Comment (sgrund14) [reply src/components/SeldonImage/SeldonImage.tsx:118]: img itself default to `eager` and `auto` respectively

**Review by scottdickerson — APPROVED** [review]
LGTM

**Pull request merged**